containers/runtime/xds/config/default_config.json

```diff
@@ -0,0 +1,217 @@
+                              }
+                           }
+                        ]
+                     }
+                  ]
+               },
+               "TTL": null
+            },
+            "default_testGrpcListenerName": {
+               "Resource": {
+                  "@type": "type.googleapis.com/envoy.config.listener.v3.Listener",
+                  "name": "default_testGrpcListenerName",
+                  "address": {
+                     "socketAddress": {
+                        "address": "0.0.0.0",
+                        "portValue": 10000
+                     }
+                  },
+                  "filterChains": [
+                     {
+                        "filters": [
+                           {
+                              "name": "envoy.filters.network.http_connection_manager",
+                              "typedConfig": {
+                                 "@type": "type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager",
+                                 "rds": {
+                                    "configSource": {
+                                       "ads": {}
+                                    },
+                                    "routeConfigName": "default_TestRouteName"
+                                 },
+                                 "httpFilters": [
+                                    {
+                                       "name": "router",
+                                       "typedConfig": {
+                                          "@type": "type.googleapis.com/envoy.extensions.filters.http.router.v3.Router"
+                                       }
+                                    }
+                                 ]
+                              }
+                           }
+                        ],
+                        "name": "filter-chain-name"
+                     }
+                  ],
+                  "apiListener": {
+                     "apiListener": {
+                        "@type": "type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager",
+                        "rds": {
+                           "configSource": {
+                              "ads": {}
+                           },
+                           "routeConfigName": "default_TestRouteName"
+                        },
+                        "httpFilters": [
+                           {
+                              "name": "router",
+                              "typedConfig": {
+                                 "@type": "type.googleapis.com/envoy.extensions.filters.http.router.v3.Router"
+                              }
+                           }
+                        ]
+                     }
+                  }
+               },
+               "TTL": null
+            }
+         }
+      },
+      {
+         "Version": "",
```

Comment:
There are 8 type of resources: Endpoint, Cluster, Route, ScopedRoute, Listener, Secret, Runtime, ExtensionConfig. In this default config, we have Endpoint, Cluster, Route and Listener, the empty fields are resource of Secret, Runtime, ScopedRoute and ExtensionConfig. Although we don't include these in the default config, the marshaller and unmarshaller have the ability to fill in these resource if the user did supply them, test will also cover these resources.

Reply:
For VersionMap field, it is used internally. not really part of the config I believe.
https://pkg.go.dev/github.com/envoyproxy/go-control-plane@v0.10.0/pkg/cache/v3#Snapshot

Reply:
What would happen if these empty objects were removed from this file? Would it still work correctly?

Reply:
I have update the code to make sure they work correctly with or without submitting the placeholders.

Reply:
Gotcha. It's just kind of confusing that there are empty objects here without any indication of what they do. It would be nice if we could add comments to this to explain.

Is it possible to add the `@type` field without filling in the rest of the object? If so, let's do that. If not, this is fine as it is.

Reply:
Thank you so much for pointing this out. I have eliminate the empty objects in the default config. :)